src/app/common/components/MustGatherBtn.tsx

```diff
@@ -31,12 +34,27 @@ export const MustGatherBtn: React.FunctionComponent<IMustGatherBtn> = ({ display
       }
     >
       <Button
+        aria-label={`Download logs for ${displayName}`}
         isAriaDisabled={!mustGathersQuery.isSuccess}
         variant="secondary"
-        component="a"
-        target="_blank"
-        download
-        href={getMustGatherApiUrl(`must-gather/${mustGather?.['id']}/data`)}
+        onClick={() => {
+          authorizedFetch<Blob>(
+            getMustGatherApiUrl(`must-gather/${mustGather?.['id']}/data`),
+            fetchContext,
+            {},
+            'get',
+            'blob'
+          )
+            .then((tarBall) => {
+              const file = new File([tarBall], mustGather['archive-name'], {
+                type: 'text/plain;charset=utf-8',
+              });
+              saveAs(file);
+            })
+            .catch((error) => {
+              console.error(error);
```

Comment:
We may want to pop a push notification in this catch statement in case it fails to download the tarball or there is some exception in file-saver (malformed blob)

Reply:
Good call - done!